### ISSUE TITLE
How to add default include directories for library found with compiler.find_library

### ISSUE THREAD

**Issue by justyn** [mesonbuild/meson#4486]
I need to link against a library installed to `/usr/local/lib` without pkgconfig.

I find the library using `compiler.find_library('libname')` and pass this as a dependency to the executable.

However the library also has headers installed to `/usr/local/include`. I currently add these with `include_directories('/usr/local/include')` passed to the executable.

Is there a way to do this without hard-coding the header directory path?

**Comment by NickeZ** [comment]
You can have it as a [build option](http://mesonbuild.com/Build-options.html) and let the user set it with `-Dname_include_path=/usr/local/include` (assuming your lib was named name...)

**Comment by dcbaker** [comment]
Did you have to do anything special to get `compiler.find_library` to find the library?

**Comment by justyn** [comment]
@NickeZ thank you but I'm trying to find a way to avoid specifying the path manually if possible

@dcbaker no, compiler.find_library finds it without any extra options

**Comment by chrishecker** [comment]
Is there a good solution to this?  Right now I have this:
```
r = run_command('dialog-config','--cflags', check: true)
platform_dep += cc.find_library('dialog', has_headers: ['dialog.h'], header_args: r.stdout().strip(), required: true)
```
which works and finds both the dialog lib and header:
```
Has header "dialog.h" : YES (cached)
Library dialog found: YES
```
but adding the generated dep from this doesn't include the header directory.  I can take the output from `dialog-config --cflags` and try and pass it to include_directories, but yuck, I'll have to remove the -I and worry about if more stuff is added to that ouput.  (I don't know why dialog doesn't use pkg-config but this is the world we live in.)

Anyway, what's the best way to do this?  This issue was never really answered...the build option is not a good solution.

Thanks,
Chris

**Comment by eli-schwartz** [comment]
You could construct a `dialog_dep = declare_dependency(compile_args: ra.split(' '), link_with: cc.find_library('dialog'))` and just pass all cflags that dialog wants to targets that link to it.

Another possible option is to add dialog support to Meson itself, via https://mesonbuild.com/Dependencies.html#config-tool.

dialog is companion software to ncurses, both are maintained by Thomas Dickey. ncurses has pkg-config support, quite possibly because it is much more popular and people have requested it. Perhaps no one has asked for a `dialog.pc`? I suspect the maintainer would be willing to add one though...

**Comment by chrishecker** [comment]
Ah, okay, that seems a bit cleaner (although the string split(' ') is still gross, but I guess if somebody has some new insane ; separator for args in the future we'll just have to add a args_split() or whatever), I'll try that.  I get that adding a builtin for dialog would work (and is kind of the "meson way", see our other discussions), I just think it's good to make sure meson can do reasonably common things in script without hacking the python.

**Comment by eli-schwartz** [comment]
Actually good point, it doesn't, I meant to say dependencies.

Because dependencies can, in turn, have other dependencies.

**Comment by chrishecker** [comment]
Yes, I was about to come here and post that after trying it.  :)
